addons/barcodes/migrations/12.0.2.0/pre-migration.py

```diff
@@ -14,15 +15,31 @@ def migrate(env, version):
             ],
         )
 
-        # done here instead of post-migration to avoid the post-init-hook
+        # done here instead of post-migration to avoid the default
         if openupgrade.table_exists(cr, 'stock_picking_type'):
-            cr.execute(
-                """
+            openupgrade.logged_query(
+                cr, """
                 UPDATE res_company rc
                 SET nomenclature_id = spt.barcode_nomenclature_id
                 FROM stock_picking_type spt
                 LEFT JOIN stock_warehouse sw ON spt.warehouse_id = sw.id
                 WHERE sw.company_id = rc.id
-                    AND spt.barcode_nomenclature_id IS NOT NULL
-                """
+                    AND spt.barcode_nomenclature_id IS NOT NULL"""
+            )
+
+        cr.execute(
+            """
+            SELECT bn.id
+            FROM ir_model_data imd
+            JOIN barcode_nomenclature bn ON (imd.res_id = bn.id AND imd.model = 'barcode.nomenclature')
+            WHERE imd.module = 'barcodes' AND imd.name = 'default_barcode_nomenclature'"""
```

Comment:
A more elegant SQL is:
```suggestion
            SELECT imd.res_id
            FROM ir_model_data imd
            WHERE imd.model = 'barcode.nomenclature'
                AND imd.module = 'barcodes'
                AND imd.name = 'default_barcode_nomenclature'"""
```
but you can use totally safe in pre-migration this:

```python
env["ir.model.data"].xmlid_to_res_id(
    "barcodes.default_barcode_nomenclature",
    raise_if_not_found=False,
)
```

Reply:
No, that give an error, because model `barcode.nomenclature` still is not loaded in the env. I just checked that in the beginning, before doing the force pushes.

Reply:
That method is not loading the record. Just returning the `ID`. See the implementation here: https://github.com/odoo/odoo/blob/7aaad8480dde97b2d04b4148e4e54beb57b7f141/odoo/addons/base/models/ir_model.py#L1409

Reply:
Oh, sorry, I was confused with `xmlid_to_object` method.